### PR TITLE
Add UI for example start rules, needed examples 

### PR DESCRIPTION
**Pull request by sakekasi**

### CODE REVIEW

File: visualizer/exampleGenerationRequests.js [hunk 1]
```diff
@@ -18,124 +18,86 @@
...
-  ExampleRequest.prototype.onChange = function(kbevent) {
-    if (kbevent.code === 'Enter' && kbevent.ctrlKey) {
-      if (this.isValid()) {
-        this.validSubmitListener({
-          text: this.domNode.value,
-          ruleName: this.ruleName
-        });
-      }
+    if (startRule !== null) {
+      var option = Array.from(dropdown.options).find(
```
Comment (pdubroy) [inline visualizer/exampleGenerationRequests.js]: Not sure how well supported `Array.from` is yet...we probably shouldn't use it.

File: visualizer/index.html [hunk 2]
```diff
@@ -422,24 +436,17 @@
       <div id="userExampleContainer">
         <h2>Examples
           <input id="addExampleButton" type="button" value="+"></input>
-          <span id="examplesNeededDrawer" class="exampleGeneratorUI hidden">Needed Examples <span class="indicator">&#x2B05;</span></span>
+          <!-- <span id="examplesNeededDrawer" class="exampleGeneratorUI hidden">Needed Examples <span class="indicator">&#x2B05;</span></span> -->
```
Comment (pdubroy) [inline visualizer/index.html]: Just delete this.

File: visualizer/index.html [hunk 3]
```diff
@@ -395,6 +395,20 @@
         outline: none;
       }
     }
+
+    #exampleBottom {
```
Comment (pdubroy) [inline visualizer/index.html]: I think this should go in examples.css.

File: visualizer/style/examples.css [hunk 4]
```diff
@@ -48,15 +51,17 @@
   color: #d44950;
   padding: 4px;
   cursor: default;
-  display: none;
+  /*display: none;*/
```
Comment (pdubroy) [inline visualizer/style/examples.css]: Delete.

File: visualizer/style/examples.css [hunk 5]
```diff
@@ -145,3 +150,49 @@
...
+}
+
+span.code {
+  flex: 1;
+}
+
+
+#neededExamples > ul {
+  list-style-type: none;
+  font-family: Menlo, Monaco, sans-serif;
```
Comment (pdubroy) [inline visualizer/style/examples.css]: If you don't want monospace you shouldn't use Meno/Monaco :-)

File: visualizer/exampleGenerationRequests.js (hunk 1 above)
Comment (sakekasi) [inline visualizer/exampleGenerationRequests.js]: done

File: visualizer/index.html (hunk 3 above)
Comment (sakekasi) [inline visualizer/index.html]: done

File: visualizer/index.html (hunk 2 above)
Comment (sakekasi) [inline visualizer/index.html]: done

File: visualizer/style/examples.css (hunk 5 above)
Comment (sakekasi) [inline visualizer/style/examples.css]: done

File: visualizer/style/examples.css (hunk 4 above)
Comment (sakekasi) [inline visualizer/style/examples.css]: done

File: visualizer/exampleGenerationRequests.js [hunk 6]
```diff
@@ -92,7 +92,7 @@
     });
 
     if (startRule !== null) {
-      var option = Array.from(dropdown.options).find(
+      var option = Array.prototype.slice.call(dropdown.options).find(
```
Comment (pdubroy) [inline visualizer/exampleGenerationRequests.js]: Oh, this is actually making an unnecessary copy -- just use `Array.prototype.find.call(dropdown.options, function(option) { ... })`

**Pull request merged**